cfme/tests/containers/test_autotagging_providers.py

```diff
@@ -0,0 +1,198 @@
+from collections import namedtuple
+import fauxfactory
+import pytest
+from random import choice
+from traceback import format_exc
+
+from cfme.configure.configuration.region_settings import MapTags
+from cfme.containers.provider import ContainersProvider
+from cfme.containers.node import Node
+from cfme.containers.replicator import Replicator
+from cfme.containers.image import Image
+from cfme.containers.pod import Pod
+from cfme.containers.project import Project
+from cfme.containers.route import Route
+
+from utils import testgen
+from utils.version import current_version
+from utils.wait import wait_for
+from utils.log import logger
+
+
+pytestmark = [
+    pytest.mark.uncollectif(lambda: current_version() < "5.8"),
+    pytest.mark.usefixtures('setup_provider_modscope'),
+    pytest.mark.tier(1)]
+pytest_generate_tests = testgen.generate([ContainersProvider], scope='module')
+
+TEST_OBJECTS = (Pod, Route, Node, Replicator, Project)
+
+values = ['{}{}'.format('value_', fauxfactory.gen_alpha(5).lower()),
+          '{}{}'.format('value_', fauxfactory.gen_alphanumeric(5).lower()),
+          '{}{}{}'.format('value_', fauxfactory.gen_alpha(3).lower(),
+                          fauxfactory.gen_alphanumeric(3))]
+
+
+def check_smart_management_in_ui(instance, expected_value):
+    return instance.summary.smart_management.my_company_tags[0].value == str(expected_value)
+
+
+def set_label(test_obj, instance, label_name, value):
+    try:
+        status_code, json_content = test_obj.set_label(instance, label_name, value)
+        # temp debugging, remove later
+        print('setting label status code is: ' + str(status_code))
+    except:
+        status_code, json_content = None, format_exc()
+    return status_code, json_content
```

Comment:
**OBSERVATION:** It's a shame we have to wrap ```set_label``` with another ```set_label``` we should discuss how to make this uniform moving forward